### PR TITLE
[CPDLP-2396] Tidy up participant declarations API spec

### PR DESCRIPTION
**Pull request by leandroalemao**
### Context

- Ticket: [CPDLP-2396](https://dfedigital.atlassian.net/browse/CPDLP-2396)

### Changes proposed in this pull request

Review declarations in the API spec and component schemas, for all API versions, and try to make sure they are consistent.

[CPDLP-2396]: https://dfedigital.atlassian.net/browse/CPDLP-2396?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ethax-ross — APPROVED** [review]
Looks great 👏 only noticed a couple minor differences, not sure if they are issues or not:

- The schema names have changed in a few places (e.g. ParticipantDeclaration -. ParticipantDeclarationRequest)
- Endpoint added (I assume it was just missing from our current docs `/api/v2/participant-declarations.csv`)
- ECFParticipantDeclarationsRequestAttributes appears to be missing (see screenshot)

Screenshot 2023-09-11 at 10 43 47

File: swagger/v2/api_spec.json [hunk 1]
```diff
@@ -66,24 +66,10 @@
         ],
         "responses": {
           "200": {
-            "description": "A list of NPQ applications",
```
Comment (cwrw) [inline swagger/v2/api_spec.json:69]: could you add those back please, as we no longer can access the [npq application](https://cpd-ecf-review-3882-web.test.teacherservices.cloud/api-reference/reference-v2.html#api-v2-npq-applications-get) response?

File: swagger/v2/api_spec.json [hunk 2]
```diff
@@ -168,24 +154,10 @@
         ],
         "responses": {
           "200": {
-            "description": "A single NPQ application",
-            "content": {
-              "application/json": {
-                "schema": {
-                  "$ref": "#/components/schemas/NPQApplicationResponse"
```
Comment (cwrw) [inline swagger/v2/api_spec.json:175]: same here and everywhere else please

File: swagger/v2/api_spec.json [hunk 3]
```diff
@@ -3259,35 +3051,20 @@
             "example": [
               {
                 "id": "db3a7848-7308-4879-942a-c4a70ced400a",
-                "type": "npq_application",
```
Comment (cwrw) [inline swagger/v2/api_spec.json:3262]: I think this needs to stay as NPQ application as its under `"MultipleNPQApplicationsResponse`, same for attributes below?

File: swagger/v2/api_spec.json [hunk 4]
```diff
@@ -5242,10 +5031,7 @@
               "retained-2",
               "retained-3",
               "retained-4",
-              "completed",
-              "extended-1",
```
Comment (cwrw) [inline swagger/v2/api_spec.json:5246]: could we keep extended types here?

File: swagger/v3/component_schemas/ECFParticipantDeclarationResponse.yml [hunk 5]
```diff
@@ -1,20 +0,0 @@
-description: "An ECF participant declaration response"
```
Comment (cwrw) [inline swagger/v3/component_schemas/ECFParticipantDeclarationResponse.yml:1]: could we also delete this from the v3/api_spec please?

**Review by cwrw — COMMENTED** [review]
Thank you @leandroalemao for tackling cleaning this up, its looking much better and cleaner now, a few comments happy to chat through but otherwise looking good

**Comment by leandroalemao** [comment]
Hi @cwrw & @ethax-ross tks for your review.. I've managed to review all comments.. the only one I had to keep is `ParticipantDeclaration` as it's being used on `SchemaValidator` so I left as it's.
When running `swaggerize`, the idea is to get no changes to api spec json files.

File: swagger/v1/api_spec.json [hunk 6]
```diff
@@ -3398,35 +3260,23 @@
             "example": [
               {
                 "id": "db3a7848-7308-4879-942a-c4a70ced400a",
-                "type": "npq_application",
+                "type": "participant",
```
Comment (cwrw) [inline swagger/v1/api_spec.json:3263]: Do you mind undoing this change as it's overwriting the correct response with the incorrect "participant" responce, as this is an NPQ application endpoint? see https://cpd-ecf-review-3882-web.test.teacherservices.cloud/api-reference/reference-v1.html#api-v1-npq-applications-get-parameters

Comment (leandroalemao) [reply swagger/v1/api_spec.json]: Hi @cwrw tks for spotting this.. I've fixed the issue .. `MultipleNPQApplicationsResponse` & `NPQApplicationAttributes` files had the incorrect attributes. I hope it's all good.. tksss

File: swagger/v3/component_schemas/ECFParticipantTransfers.yml [hunk 7]
```diff
@@ -28,18 +28,16 @@ properties:
       - complete
   leaving:
     nullable: true
-    $ref: "#/components/schemas/ECFParticipantLeaving"
-    example:
-      school_urn: "123456"
```
[the remark this answers is not in the document]
Comment (leandroalemao) [reply swagger/v3/component_schemas/ECFParticipantTransfers.yml:33]: Hi @cwrw tks for your review.. it's all fixed.. nested schemas are being displayed now.. 

https://cpd-ecf-review-3882-web.test.teacherservices.cloud/api-reference/reference-v3.html#schema-ecfparticipanttransfers
https://cpd-ecf-review-3882-web.test.teacherservices.cloud/api-reference/reference-v3.html#schema-ecfenrolment
https://cpd-ecf-review-3882-web.test.teacherservices.cloud/api-reference/reference-v3.html#schema-npqenrolment

**Review by cwrw — APPROVED** [review]
Thank you @leandroalemao for all this! well done ⭐

**Pull request merged**